cng/tls1prf.go

```diff
@@ -73,11 +76,13 @@ func TLS1PRF(secret, label, seed []byte, keyLen int, h func() hash.Hash) ([]byte
 		Count:   uint32(len(buffers)),
 		Buffers: &buffers[0],
 	}
-	out := make([]byte, keyLen)
 	var size uint32
-	err = bcrypt.KeyDerivation(kh, params, out, &size, 0)
+	err = bcrypt.KeyDerivation(kh, params, result, &size, 0)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return out[:size], nil
+	if size != uint32(len(result)) {
```

Comment:
```suggestion
        // If the generated size doesn't match the result slice length, the entropy limit might be reached.
	if size != uint32(len(result)) {
```

Reply:
I agree that this block deserved some comments. Your proposal describes what the code does, but not the why. Also, the error message is misleading, there could be other causes to return less bytes than requested. Better return a more general error in such case.

What about this:

```go
// The Go standard library expects TLS1PRF to return the requested number of bytes,
// fail if it doesn't.
if size != uint32(len(result)) {
	return errors.New("tls1-prf: derived less bytes than requested")
}
```

Reply:
Yeah sounds good